### PR TITLE
first testversion

### PR DESCRIPTION
**Pull request by Gerbenvandervries**

### CODE REVIEW

File: protocols/Bowtie2.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+-x ${DudesReference} \
+--no-unal \
+--fast \
+-p 6 \
+-k 50 \
+-1 ${sampleKneadDataOutPE1} \
+-2 ${sampleKneadDataOutPE2} \
+-S ${tmpsampleBowtie2Sam}
+
+mv ${tmpsampleBowtie2Sam} ${sampleBowtie2Sam}
```
Comment (RoanKanninga) [inline protocols/Bowtie2.sh]: add echo "mv ${tmpsampleBowtie2Sam} ${sampleBowtie2Sam}"

File: protocols/CreateExternSamplesProjects.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#
+# Create project dirs.
+#
+mkdir -p ${projectRawtmpDataDir}
+mkdir -p ${projectJobsDir}
+mkdir -p ${intermediateDir}
+mkdir -p ${projectResultsDir}
+mkdir -p ${projectQcDir}
+
+ROCKETPOINT=`pwd`
```
Comment (RoanKanninga) [inline protocols/CreateExternSamplesProjects.sh]: would prefer $(pwd)

File: protocols/Humann2.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#string Bowtie2Version
+#string humann2Version
+#string sampleMetaphlanOutTxt
+#string project
+#string group
+#string tmpDirectory
+#string logsDir
+
+makeTmpDir ${sampleKneadDataOut}
+tmpsampleKneadDataOut=${MC_tmpFile}
```
Comment (RoanKanninga) [inline protocols/Humann2.sh]: unused, i guess

File: protocols/Kneaddata.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+--bowtie2 ${EBROOTBOWTIE2}/bin/ \
+--fastqc ${EBROOTFASTQC} \
+--output ${tmpsampleKneadDataOut}/ \
+--log ${tmpsampleKneadDataOut}/${externalSampleID}.log
+
+cat ${tmpsampleKneadDataOut}/${externalSampleID}_1_kneaddata_paired_1.fastq ${tmpsampleKneadDataOut}/${externalSampleID}_1_kneaddata_paired_2.fastq > ${tmpsampleKneadDataOut}/${externalSampleID}.kneaddata.merged.fastq
+mv ${tmpsampleKneadDataOut}/${externalSampleID}.kneaddata_merged.fastq ${sampleKneadDataOut}/${externalSampleID}.kneaddata.merged.fastq
+mv ${tmpsampleKneadDataOut}/${externalSampleID}_1_kneaddata_paired_1.fastq ${sampleKneadDataOut}
+mv ${tmpsampleKneadDataOut}/${externalSampleID}_1_kneaddata_paired_2.fastq ${sampleKneadDataOut}
+mv ${tmpsampleKneadDataOut}/${externalSampleID}.log ${sampleKneadDataOut}
```
Comment (RoanKanninga) [inline protocols/Kneaddata.sh]: adding print statement

File: protocols/Metaphlan2.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+echo "Starting taxonomy classification using Metaphlan"
+metaphlan2.py ${sampleKneadDataMergedFasta} \
+--input_type multifastq \
+--mpa_pkl ${MetaPhlAn2Index} \
+--nproc 6 \
+-o ${tmpsampleMetaphlanOutDir}/${externalSampleID}_metaphlan.txt \
+--tmp_dir ${tmpsampleMetaphlanOutDir}
+
+mv ${tmpsampleMetaphlanOutDir}/${externalSampleID}_metaphlan.txt ${sampleMetaphlanOutTxt}
```
Comment (RoanKanninga) [inline protocols/Metaphlan2.sh]: add print statement

File: protocols/TrimReads.sh [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+#string project
+#string intermediateDir
+#string project
+#string group
+#string tmpDirectory
+#string logsDir
+
+#Echo parameter values
+echo "seqType: ${seqType}"
+echo "peEnd1BarcodeFqGz: ${peEnd1BarcodeFqGz}"
```
Comment (RoanKanninga) [inline protocols/TrimReads.sh]: variables ${peEnd1BarcodeFqGz}, ${srBarcodeFqGz} and ${peEnd2BarcodeFqGz} are not included above

File: protocols/TrimReads.sh [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	ktrim=l \
+        qtrim=rl \
+        trimq=14 \
+        minlength=20 > ${intermediateDir}/${externalSampleID}.trimming.log 2>&1
+	#\forcetrimleft=11
+
+	gzip ${fastq1}.tmp
+	gzip ${fastq2}.tmp
+	mv ${fastq1}.tmp.gz ${peEnd1BarcodeFqGz}
+	mv ${fastq2}.tmp.gz ${peEnd2BarcodeFqGz}
```
Comment (RoanKanninga) [inline protocols/TrimReads.sh]: you can also do:
gzip -c ${fastq1}.tmp > ${peEnd1BarcodeFqGz}
rm ${fastq1}.tmp

File: templates/generate_template.sh [hunk 8]
```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+#module load NGS_Microbiome
+module load Molgenis-Compute/v17.08.1-Java-1.8.0_74
+module list
+
+EBROOTNGS_MICROBIOME="/home/umcg-gvdvries/github/NGS_Microbiome/"
```
Comment (RoanKanninga) [inline templates/generate_template.sh:7]: I guess this should be changed after testing ;)

**Comment by RoanKanninga** [comment]
You should add quotes around all the variables for safety

File: protocols/Bowtie2.sh [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+#Load module
+module load "${Bowtie2Version}"
+module list
+
+makeTmpDir "${sampleBowtie2Sam}"
+tmpsampleBowtie2Sam=${MC_tmpFile}
+
+${EBROOTBOWTIE2}/bin/bowtie2 \
+-x "${DudesReference}" \
```
Comment (RoanKanninga) [inline protocols/Bowtie2.sh]: misschien netter om variabelen niet met een hoofdletter te laten beginnen

**Review by marieke-bijlsma — APPROVED** [review]

**Pull request merged**